src/main/java/net/minecraftforge/event/entity/player/AdvancementEvents.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2022.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.entity.player;
+
+import net.minecraft.advancements.Advancement;
+import net.minecraft.advancements.AdvancementProgress;
+import net.minecraft.world.entity.player.Player;
+import net.minecraftforge.common.MinecraftForge;
+
+public class AdvancementEvents
+{
+    /**
+     * This event is fired when a player earns an advancement/recipe.
+     * <br>
+     * This event is not {@link net.minecraftforge.eventbus.api.Cancelable}.<br>
+     * <br>
+     * This event does not have a result. {@link HasResult}<br>
+     * <br>
+     * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
+     */
+    public static class AdvancementEarnEvent extends PlayerEvent
+    {
+        private final Advancement advancement;
+
+        public AdvancementEarnEvent(Player player, Advancement earned)
+        {
+            super(player);
+            this.advancement = earned;
+        }
+
+        public Advancement getAdvancement()
+        {
+            return advancement;
+        }
+    }
+
+    /**
+     * This event is fired when a player progresses on an advancement criterion.
+     * <br>
+     * This event is not {@link net.minecraftforge.eventbus.api.Cancelable}.<br>
+     * <br>
+     * This event does not have a result. {@link HasResult}<br>
+     * <br>
+     * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
+     */
+    public static class AdvancementProgressEvent extends PlayerEvent
+    {
+        public enum ProgressType
+        {
+            GRANT,REVOKE
```

Comment:
Formatting: space after comma between enum value declarations:
```suggestion
            GRANT, REVOKE
```